libs/geo/src/main/java/org/elasticsearch/geo/utils/WellKnownText.java

```diff
@@ -0,0 +1,557 @@
+                    } else {
+                        sb.append(LPAREN);
+                        visitPoint(point.lon(), point.lat());
+                        sb.append(RPAREN);
+                    }
+                    return null;
+                }
+
+                private void visitPoint(double lon, double lat) {
+                    sb.append(lon).append(SPACE).append(lat);
+                }
+
+                private void visitCollection(GeometryCollection<?> collection) {
+                    if (collection.size() == 0) {
+                        sb.append(EMPTY);
+                    } else {
+                        sb.append(LPAREN);
+                        collection.get(0).visit(this);
+                        for (int i = 1; i < collection.size(); ++i) {
+                            sb.append(COMMA);
+                            collection.get(i).visit(this);
+                        }
+                        sb.append(RPAREN);
+                    }
+                }
+
+                @Override
+                public Void visit(Polygon polygon) {
+                    sb.append(LPAREN);
+                    visit((Line) polygon.getPolygon());
+                    int numberOfHoles = polygon.getNumberOfHoles();
+                    for (int i = 0; i < numberOfHoles; ++i) {
+                        sb.append(", ");
+                        visit((Line) polygon.getHole(i));
+                    }
+                    sb.append(RPAREN);
+                    return null;
+                }
+
+                @Override
+                public Void visit(Rectangle rectangle) {
+                    sb.append(LPAREN);
+                    // minX, maxX, maxY, minY
+                    sb.append(rectangle.getMinLon());
+                    sb.append(COMMA);
+                    sb.append(SPACE);
+                    sb.append(rectangle.getMaxLon());
+                    sb.append(COMMA);
+                    sb.append(SPACE);
+                    sb.append(rectangle.getMaxLat());
+                    sb.append(COMMA);
+                    sb.append(SPACE);
+                    sb.append(rectangle.getMinLat());
+                    sb.append(RPAREN);
+                    return null;
+                }
+            });
+        }
+    }
+
+    public static Geometry fromWKT(String wkt) throws IOException, ParseException {
+        StringReader reader = new StringReader(wkt);
+        try {
+            // setup the tokenizer; configured to read words w/o numbers
+            StreamTokenizer tokenizer = new StreamTokenizer(reader);
+            tokenizer.resetSyntax();
+            tokenizer.wordChars('a', 'z');
+            tokenizer.wordChars('A', 'Z');
+            tokenizer.wordChars(128 + 32, 255);
+            tokenizer.wordChars('0', '9');
+            tokenizer.wordChars('-', '-');
+            tokenizer.wordChars('+', '+');
+            tokenizer.wordChars('.', '.');
+            tokenizer.whitespaceChars(0, ' ');
```

Comment:
Why is `0` a whitespaceChar ?

Reply:
Not only 0. Basically this line means that everything between 0 and ` ` will be treated as a whitespace character. This is to ignore all control characters below space ` ` including new line,  line feed, tabs, new page etc. by treating it as a whitespace character. I have decided to leave it as is instead of configuring each a subset of character manually since this is [the default](https://docs.oracle.com/javase/7/docs/api/java/io/StreamTokenizer.html#StreamTokenizer(java.io.InputStream)) behavior of StreamTokenizer, but we can be more restrictive here and allow only spaces, tabs and line feeds as suggested in the [standard](http://docs.opengeospatial.org/is/12-063r5/12-063r5.html#138) @iverase, @nknize what do you think?

Reply:
The standard seems to only allow spaces, tabs or line feeds to improve readability so let's do as recommended.

Reply:
:+1: